src/programs/timeout-evasion.ts

```diff
@@ -0,0 +1,102 @@
+import { Command, CommandHandler, DiscordEvent } from "../event-distribution";
+import { GuildMember, PartialGuildMember } from "discord.js";
+import { textLog } from "../common/moderator";
+import Tools from "../common/tools";
+import prisma from "../prisma";
+import { createYesBotLogger } from "../log";
+
+const logger = createYesBotLogger("program", "timeout-evasion");
+
+const isUserTimedOut = async (
+  member: GuildMember | PartialGuildMember
+): Promise<boolean> => {
+  return !!(await prisma.timedOutUsers.findFirst({
+    where: {
+      userId: member.id,
+    },
+  }));
+};
+
+@Command({
+  event: DiscordEvent.GUILD_MEMBER_UPDATE,
+  roleNamesAdded: ["Time Out"],
+  description:
+    "This handler adds the user to the DB when he gains the time out role",
+})
+class TimeOutAdded implements CommandHandler<DiscordEvent.GUILD_MEMBER_UPDATE> {
+  async handle(member: GuildMember): Promise<void> {
+    if (await isUserTimedOut(member)) return;
+
+    const engineerRole = Tools.getRoleByName(
+      process.env.ENGINEER_ROLE_NAME,
+      member.guild
+    );
+    try {
+      await prisma.timedOutUsers.create({
+        data: {
+          userId: member.id,
+        },
+      });
+    } catch (e) {
+      logger.error("Failed to add user to the DB " + e);
```

Comment:
If `e` is an object, this will end up as `"Failed to add user to the DB [object object]"`. You can pass the error as second argument to have it properly show in the log output:

```suggestion
      logger.error("Failed to add user to the DB ", e);
```